src/components/login/Login.js

```diff
@@ -3,65 +3,92 @@ import { useState } from "react";
 import { useAuth } from "../../contexts/auth-context";
 import axios from "axios";
 import { Link } from "react-router-dom";
-
-const Login = () => {
-  const [userData, setUserData] = useState({ email: "", password: "" });
+import { validLogin } from "../../utilities/auth";
+const UserLogin = () => {
+  const [loginData, setLoginData] = useState({ email: "", password: "" });
   const { authState, authDispatch } = useAuth();
-
+  const [loginErrors, setLoginErrors] = useState({
+    email: "",
+    password: "",
+  });
   const inputHandler = (e) => {
-    setUserData((data) => ({ ...data, [e.target.name]: e.target.value }));
+    setLoginData((data) => ({ ...data, [e.target.name]: e.target.value }));
+    setLoginErrors((loginErr) => ({ ...loginErr, [e.target.name]: "" }));
   };
 
-  const postLoginDetails = async ({ email, password }) => {
-    authDispatch({ type: "USER_LOAD" });
+  const postLoginDetails = async (email, password) => {
     try {
-      console.log(email, password);
-      let { data } = await axios.post("/api/auth/login", { email, password });
-      authDispatch({ type: "USER_LOAD_SUCCESS", payload: data });
+      authDispatch({ type: "USER_LOAD" });
+      const { data } = await axios.post("/api/auth/login", { email, password });
+      authDispatch({ type: "USER_LOAD_SUCCESS", payload: data.foundUser });
       console.log(data);
       localStorage.setItem("token", data.encodedToken);
     } catch (err) {
       console.log(err);
     }
   };
   console.log(authState);
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const { isValid, errors } = validLogin(loginData, loginErrors);
+    if (!isValid) {
+      setLoginErrors(errors);
+      return;
+    }
+    postLoginDetails(loginData.email, loginData.password);
+  };
+
   return (
-    <main className="bg">
-      <div class="login-container center">
-        <h1 class="login-title">Login</h1>
+    <form className="bg" onSubmit={handleSubmit}>
+      <div className="login-container center">
+        <h1 className="login-title">Login</h1>
 
         <input
           type="text"
           placeholder="Email"
           name="email"
-          value={userData.email}
+          value={loginData.email}
```

Comment:
you can destructure login data and extract out email and password

Reply:
ok